internal/pkg/component/gitlab/gitlab.go

```diff
@@ -4,12 +4,15 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"net/url"
 	"strings"
 
 	appstudiov1alpha1 "github.com/konflux-ci/application-api/api/v1alpha1"
 	"github.com/konflux-ci/mintmaker/internal/pkg/component/base"
 	bslices "github.com/konflux-ci/mintmaker/internal/pkg/slices"
 	"github.com/konflux-ci/mintmaker/internal/pkg/utils"
+	gitlab_mintmaker "github.com/konflux-ci/mintmaker/pkg/git/gitlab"
```

Comment:
we're going to remove the `pkg` directory (not the `internal/pkg`) as a code clean up step in this refactoring work, could you move the necessary APIs to `internal/pkg/component/gitlab`?

Reply:
good point. Rebased.